redbeams/src/main/java/com/sequenceiq/redbeams/service/network/SubnetChooserService.java

```diff
@@ -18,20 +21,25 @@
 @Service
 public class SubnetChooserService {
 
+    private static final Logger LOGGER = LoggerFactory.getLogger(SubnetChooserService.class);
+
     @Inject
     private CloudPlatformConnectors cloudPlatformConnectors;
 
     public List<CloudSubnet> chooseSubnets(List<CloudSubnet> subnetMetas, CloudPlatform cloudPlatform, DBStack dbStack) {
-        SubnetSelectionResult subnetSelectionResult =
-                cloudPlatformConnectors.get(new CloudPlatformVariant(dbStack.getCloudPlatform(), dbStack.getPlatformVariant()))
-                .networkConnector()
-                .selectSubnets(
-                        subnetMetas,
-                        SubnetSelectionParameters.builder()
-                                .withHa(dbStack.isHa())
-                                .withForDatabase()
-                                .build()
-                );
+        NetworkConnector networkConnector = cloudPlatformConnectors.get(new CloudPlatformVariant(dbStack.getCloudPlatform(), dbStack.getPlatformVariant()))
+                .networkConnector();
+        SubnetSelectionParameters build = SubnetSelectionParameters
+                .builder()
+                .withHa(dbStack.isHa())
+                .withPreferPrivateIfExist()
+                .build();
+        SubnetSelectionResult subnetSelectionResult = networkConnector
+                .filterSubnets(subnetMetas, build);
+        LOGGER.info("Filtered subnets are: {}", subnetSelectionResult);
```

Comment:
I think there is enough logging within the selection logic, I would not list the result here again. If logging in the filter - selection is not enough, I would add it there.